### ISSUE TITLE
Golangci-lint does not work in Golang Workspace

### ISSUE THREAD

**Issue by rajkumarGosavi** [golangci/golangci-lint#3798]
Hi,

I am still facing the issue in 1.52.x version of golangci-lint.

In my case it does not give any lint errors even though there are many.

How my project is structured

```
A
|-- go.mod
|-- go.sum
|-- .golangci.yml

B
|-- go.mod
|--go.sum
go.work
```

And I run my golangci-lint as `cd B && golangci-lint run -c ../A/.golangci.yml`
And I get 0 lint errors.

But if I structure it without workspace

```
A
|-- go.mod
|-- go.sum
|-- .golangci.yml
```

```
B
|-- go.mod
|-- go.sum
```

And when i call `cd B && golangci-lint run -c ../A/.golangci.yml` i can see the lint errors listed as per the config.

There is no change in config in both the cases.

Only change is that the 1st one is using go workspace `go.work` and the later one is not.

_Originally posted by @rajkumarGosavi in https://github.com/golangci/golangci-lint/issues/2654#issuecomment-1518092018_

**Comment by ldez** [comment]
duplicate of #2654

**Comment by rajkumarGosavi** [comment]
Hi @ldez , 
Are we looking into the #2654 issue then? 
Cause I raised it over there but got no response and the discussions were going in a different direction. Thus opened this ticket from the same comment.

Thanks !!
